### DISCUSSION TITLE
Parameterized Dimension Support

### DISCUSSION THREAD

**Discussion by astnmsn** [webonnx/wonnx#206] [Q&A]
I am looking to use wonnx in the browser for [segment-anything](https://segment-anything.com/demo#).
The model requires that some inputs/outputs have dynamic shape and use parameterized dimensions to support this.
It is clear that wonnx has explicitly disallowed this: https://github.com/webonnx/wonnx/blob/master/wonnx/src/utils.rs#L493

Is this something that can be supported in the future?
How difficult of a task is it to integrate this?
Do you have any alternative suggestions for this?

**Comment by pixelspark** [comment]
Hi @astnmsn - currently all dimensions should have known values at inference time, as we need to know the size of buffers to allocate. However you may use the shape inference code to convert parametrized dimensions to static ones right before an inference pass. The shape inference code is there but not perfect yet - it all depends on which ops are needed.
